src/HAEntity.cpp

```diff
@@ -1,22 +1,16 @@
 
 #include "HAEntity.hpp"
+
 using std::cerr;
 using std::endl;
 
 HAEntity::HAEntity(json _state)
 {
-  cerr << _state.dump();
-
   state = _state;
 
   id = state["entity_id"];
-  if (state.contains("attributes") && state["attributes"].contains("friendly_name")) {
-    name = state["attributes"]["friendly_name"];
-  }
-  else {
-    name = state["entity_id"];
-  }
-  //   name = state["attributes"]["friendly_name"];
+  name = this->getNameFromState();
```

Comment:
I think you can leave out `this->` here?

Reply:
Yes, i don't know what's normal in c++ :-)

Reply:
leaving it out :)